### PR TITLE
Refactor to use Promise

### PR DESCRIPTION
**Pull request by katainaka0503**
Refactored some files to use Promise.

### CODE REVIEW

**Comment by bpasero** [comment]
@katainaka0503 looks like my last push introduced merge conflict, can you check?

**Comment by katainaka0503** [comment]
@bpasero fixed conflict.

File: src/vs/workbench/parts/git/node/git.lib.ts [hunk 1]
```diff
@@ -327,20 +327,14 @@ export class Repository {
...
-						fileOperationResult: FileOperationResult.FILE_IS_BINARY
-					});
-				} else {
-					c(this.doBuffer(object));
-				}
-			});
-		});
+		return detectMimesFromStream(child.stdout, null).then(result =>
+			isBinaryMime(result.mimes) ?
+				TPromise.wrapError<string>(<IFileOperationResult>{
```
Comment (bpasero) [inline src/vs/workbench/parts/git/node/git.lib.ts]: @katainaka0503 return missing here?

**Review by bpasero — CHANGES_REQUESTED** [review]
Looks good to me, except for one minor thing.

File: src/vs/workbench/parts/git/node/git.lib.ts (hunk 1 above)
Comment (katainaka0503) [inline src/vs/workbench/parts/git/node/git.lib.ts]: OK. I am fixing this. Is it better to write return explicitly?

**Comment by bpasero** [comment]
@katainaka0503 builds are failing due to bad whitespace indentation (e.g. https://travis-ci.org/Microsoft/vscode/jobs/206537447), can you format your changes and submit again?

**Comment by bpasero** [comment]
Thanks 👍

**Comment by katainaka0503** [comment]
Thank you for the merge!

**Pull request merged**